pages/project/[project_slug]/index.js

```diff
@@ -324,9 +320,31 @@ function GraphTooltip({ repository, tooltipData }) {
 }
 
 function Metrics({ id, repository }) {
+  const router = useRouter()
+  const { metric: queryKey } = router.query
   const { data: metrics } = useSWR('/api/metrics/' + id, fetcher)
   const [selectedMetric, setSelectedMetric] = useState()
-  const displayedMetric = selectedMetric || metrics?.[0]
+
+  let displayedMetric = selectedMetric
+
+  if (!displayedMetric) {
+    const queryMetric = metrics?.find((metric) => metric.key === queryKey)
+    const defaultMetric = metrics?.[0]
+
+    displayedMetric = queryMetric || defaultMetric
+  }
+
+  if (displayedMetric && queryKey !== displayedMetric.key) {
+    router.push(
```

Comment:
Not sure about pushing to the router on render? Can we update the links instead of hacking things like that?

Reply:
Good point, this should not be here. What do you mean by _update the links_?

Reply:
I mean instead of having `displayedMetric` always computed from the router instead of the opposite. If you want to change `displayedMetric`, either use a `Link` or `router.push` in the event handler.